src/PromoService.kt

```
@@ -0,0 +1,1342 @@
+    }
+
+    /**
+     * this method apply affiliate/referral promo for the new service if the Client already applied one affiliate/
+     * referral promo and which is in APPLIED State
+     *
+     * @param user
+     */
+    fun applyAffiliateOrReferralPromo(user: User, productType: ProductType): UserPromo? {
+        return try {
+            createUserPromoEntryIfEligible(user, productType)
+        } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)